app/uk/gov/hmrc/gform/translation/TextExtractor.scala

```diff
@@ -638,38 +631,181 @@ object TextExtractor {
+          }
+        } { taskList =>
+          taskList.sections.toList.zipWithIndex.flatMap { case (taskSection, sectionIndex) =>
+            taskSection.tasks.toList.zipWithIndex.flatMap { case (task, taskIndex) =>
+              task.sections.toList.zipWithIndex.map { case (section, taskSectionIndex) =>
+                PathIndex(sectionIndex, Some(taskIndex), Some(taskSectionIndex)) -> sectionToSectionType(section)
+              }
+            }
+          }
+        }.toMap
+
+      def getPathIndex(path: String): Option[PathIndex] = {
+        val sectionPattern = raw"\.sections\[(\d+)\]".r
+        val taskPattern = raw"\.tasks\[(\d+)\]".r
+
+        // Find all .sections[n] indices
+        val sectionIndices = sectionPattern.findAllMatchIn(path).map(_.group(1)).toList
+        val taskIndices = taskPattern.findAllMatchIn(path).map(_.group(1)).toList
+
+        (sectionIndices, taskIndices) match {
+          // expecting something like outer section, task, nested section
+          case (outer :: nested :: Nil, task :: Nil) =>
+            Some(PathIndex(outer.toInt, Some(task.toInt), Some(nested.toInt)))
+          case (outer :: Nil, task :: Nil) =>
+            Some(PathIndex(outer.toInt, Some(task.toInt), None))
+          case (List(outer), Nil) =>
+            Some(PathIndex(outer.toInt, None, None))
+          // fallback for unexpected cases
+          case _ =>
+            None
+        }
+      }
+
+      def getPageNumberFromPath(path: String) =
+        path
+          .split("\\.pages\\[")
+          .lastOption
+          .flatMap(_.split("]").headOption)
+          .flatMap(num => Try(num.toInt).toOption)
+
+      def containAnyOf(str: String, segments: Seq[String]) =
+        segments.exists { segment =>
+          str.contains(segment)
+        }
+
+      val summarySectionPaths = Seq(".summarySection", ".summaryDescription", ".summaryName")
+
+      def pathToSectionNumber(path: String) = {
+        val index = getPathIndex(path)
+        val sectionType = index.flatMap(pathIndexToSectionType.get)
+        val pageNumber = getPageNumberFromPath(path)
+
+        def getClassic(normalIndex: Int, sectionType: SectionType) = {
+          val index = TemplateSectionIndex(normalIndex)
+          sectionType match {
+            case ClassicNormal    => Some(Classic.NormalPage(index))
+            case ClassicRepeating => Some(Classic.RepeatedPage(index, 0))
+            case ClassicAddToList =>
+              pageNumber match {
+                case Some(pageNumber)                             => Some(Classic.AddToListPage.Page(index, 1, pageNumber))
+                case None if path.contains(".defaultPage")        => Some(Classic.AddToListPage.DefaultPage(index))
+                case None if path.contains(".cyaPage")            => Some(Classic.AddToListPage.CyaPage(index, 1))
+                case None if path.contains(".addAnotherQuestion") => Some(Classic.AddToListPage.RepeaterPage(index, 1))
```

Comment:
Path like `.sections[0].title` should also be covered by this condition. In this instance we know that title in rendered on repeater page.

(assuming 0 to be atl section)